In [44]:
#importing libraries
from sklearn import datasets
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings("ignore")

In [45]:
#reading training data set
mm = pd.read_csv('Merged_Movies_data.csv')
print("First few rows of the training dataset:")
mm.head()

First few rows of the training dataset:


movieId  imdbId   tmdbId                        title original_language  \
0        1  114709    862.0                    Toy Story                en   
1        2  113497   8844.0                      Jumanji                en   
2        3  113228  15602.0             Grumpier Old Men                en   
3        4  114885  31357.0            Waiting to Exhale                en   
4        5  113041  11862.0  Father of the Bride Part II                en   

                                             tagline  \
0                                                NaN   
1          Roll the dice and unleash the excitement!   
2  Still Yelling. Still Fighting. Still Ready for...   
3  Friends are the people who let you be yourself...   
4  Just When His World Is Back To Normal... He's ...   

                                            overview  popularity  adult  \
0  Led by Woody, Andy's toys live happily in his ...   21.946943  False   
1  When siblings Judy and Peter discover an encha...   17.015539  False   
2  A family wedding reignites the ancient feud be...   11.712900  False   
3  Cheated on, mistreated and stepped on, the wom...    3.859495  False   
4  Just when George Banks has recovered from his ...    8.387519  False   

                        poster_path  ...    status  video vote_average  \
0  /rhIRbceoE9lR4veEXuwCC2wARtG.jpg  ...  Released  False          7.7   
1  /vzmL6fP7aPKNKPRTFnZmiUfciyV.jpg  ...  Released  False          6.9   
2  /6ksm1sjKMFLbO7UY2i6G1ju9SML.jpg  ...  Released  False          6.5   
3  /16XOMpEaLWkrcPqSQqhTmeJuqQl.jpg  ...  Released  False          6.1   
4  /e64sOI48hQXyru7naBFyssKFxVd.jpg  ...  Released  False          5.7   

  vote_count                                           keywords  \
0     5415.0  [{'id': 931, 'name': 'jealousy'}, {'id': 4290,...   
1     2413.0  [{'id': 10090, 'name': 'board game'}, {'id': 1...   
2       92.0  [{'id': 1495, 'name': 'fishing'}, {'id': 12392...   
3       34.0  [{'id': 818, 'name': 'based on novel'}, {'id':...   
4      173.0  [{'id': 1009, 'name': 'baby'}, {'id': 1599, 'n...   

                                                cast  \
0  [{'cast_id': 14, 'character': 'Woody (voice)',...   
1  [{'cast_id': 1, 'character': 'Alan Parrish', '...   
2  [{'cast_id': 2, 'character': 'Max Goldman', 'c...   
3  [{'cast_id': 1, 'character': "Savannah 'Vannah...   
4  [{'cast_id': 1, 'character': 'George Banks', '...   

                                                crew  \
0  [{'credit_id': '52fe4284c3a36847f8024f49', 'de...   
1  [{'credit_id': '52fe44bfc3a36847f80a7cd1', 'de...   
2  [{'credit_id': '52fe466a9251416c75077a89', 'de...   
3  [{'credit_id': '52fe44779251416c91011acb', 'de...   
4  [{'credit_id': '52fe44959251416c75039ed7', 'de...   

                original_title    imdb_id  \
0                    Toy Story  tt0114709   
1                      Jumanji  tt0113497   
2             Grumpier Old Men  tt0113228   
3            Waiting to Exhale  tt0114885   
4  Father of the Bride Part II  tt0113041   

                                  poster_working_url  
0                                    Image not found  
1  https://image.tmdb.org/t/p/w500/vzmL6fP7aPKNKP...  
2                                    Image not found  
3                                    Image not found  
4                                    Image not found  

[5 rows x 30 columns]

In [76]:
mm.shape

(47291, 30)

In [46]:
mm_image = mm[~(mm['poster_working_url'] == 'Image not found')]
mm_image

movieId   imdbId    tmdbId                             title  \
1            2   113497    8844.0                           Jumanji   
18          19   112281    9273.0    Ace Ventura: When Nature Calls   
19          20   113845   11517.0                       Money Train   
31          32   114746      63.0                    Twelve Monkeys   
33          34   112431    9598.0                              Babe   
...        ...      ...       ...                               ...   
46894   176235   457437   14885.0  Pooh's Heffalump Halloween Movie   
46898   176249  5690142  404604.0                               Mom   
46905   176267   102797   30840.0                        Robin Hood   
46907   176271  2028550  111109.0               Century of Birthing   
46910   176279  6980792  461257.0                          Queerama   

      original_language                                    tagline  \
1                    en  Roll the dice and unleash the excitement!   
18                   en    New animals. New adventures. Same hair.   
19                   en                Get on, or GET OUT THE WAY!   
31                   en                     The future is history.   
33                   en              A little pig goes a long way.   
...                 ...                                        ...   
46894                en         Celebrate Lumpy's First Halloween.   
46898                hi                                        NaN   
46905                en                                        NaN   
46907                tl                                        NaN   
46910                en                                        NaN   

                                                overview  popularity  adult  \
1      When siblings Judy and Peter discover an encha...   17.015539  False   
18     Summoned from an ashram in Tibet, Ace finds hi...    8.205448  False   
19     A vengeful New York transit cop decides to ste...    7.337906  False   
31     In the year 2035, convict James Cole reluctant...   12.297305  False   
33     Babe is a little pig who doesn't quite know hi...   14.404764  False   
...                                                  ...         ...    ...   
46894  It's Halloween in the 100 Acre Wood, and Roo's...    2.568495  False   
46898  The bliss of a biology teacher’s family life i...    1.559596  False   
46905  Yet another version of the classic epic, with ...    5.683753  False   
46907  An artist struggles to finish his work while a...    0.178241  False   
46910  50 years after decriminalisation of homosexual...    0.163015  False   

                            poster_path  ...    status  video vote_average  \
1      /vzmL6fP7aPKNKPRTFnZmiUfciyV.jpg  ...  Released  False          6.9   
18     /wRlGnJhEzcxBjvWtvbjhDSU1cIY.jpg  ...  Released  False          6.1   
19     /jSozzzVOR2kfXgTUuGnbgG2yRFi.jpg  ...  Released  False          5.4   
31     /6Sj9wDu3YugthXsU0Vry5XFAZGg.jpg  ...  Released  False          7.4   
33     /gN6X3fwPya8pLffk9OEWV3DqBnE.jpg  ...  Released  False          6.0   
...                                 ...  ...       ...    ...          ...   
46894  /yU6byKhpzHG1RFpY5PYM6mfhfU1.jpg  ...  Released  False          5.4   
46898  /zZwbntqdfKdVgzH1RoMHa99g0mJ.jpg  ...  Released  False          6.6   
46905  /fQC46NglNiEMZBv5XHoyLuOWoN5.jpg  ...  Released  False          5.7   
46907  /xZkmxsNmYXJbKVsTRLLx3pqGHx7.jpg  ...  Released  False          9.0   
46910  /s5UkZt6NTsrS7ZF0Rh8nzupRlIU.jpg  ...  Released  False          0.0   

      vote_count                                           keywords  \
1         2413.0  [{'id': 10090, 'name': 'board game'}, {'id': 1...   
18        1128.0  [{'id': 409, 'name': 'africa'}, {'id': 1551, '...   
19         224.0  [{'id': 380, 'name': 'brother brother relation...   
31        2470.0  [{'id': 222, 'name': 'schizophrenia'}, {'id': ...   
33         756.0  [{'id': 3020, 'name': 'sheep'}, {'id': 4931, '...   
.

In [47]:
mm_image.isnull().sum()

movieId                     0
imdbId                      0
tmdbId                      0
title                       0
original_language           3
tagline                  3569
overview                  120
popularity                  0
adult                       0
poster_path                 0
budget                      0
genres                      0
homepage                 5169
belongs_to_collection    5608
production_companies        0
production_countries        0
release_date               12
revenue                     0
runtime                    31
spoken_languages            0
status                      6
video                       0
vote_average                0
vote_count                  0
keywords                    1
cast                        1
crew                        1
original_title              0
imdb_id                     4
poster_working_url          0
dtype: int64

In [48]:
movies_data = mm_image.drop(columns = ['belongs_to_collection', 'homepage', 'tagline', 'imdb_id', 'poster_path', 'original_title'] , axis = 1 )

In [49]:
movies_data.isnull().sum()

movieId                   0
imdbId                    0
tmdbId                    0
title                     0
original_language         3
overview                120
popularity                0
adult                     0
budget                    0
genres                    0
production_companies      0
production_countries      0
release_date             12
revenue                   0
runtime                  31
spoken_languages          0
status                    6
video                     0
vote_average              0
vote_count                0
keywords                  1
cast                      1
crew                      1
poster_working_url        0
dtype: int64

In [50]:
movies_data.dropna(inplace = True)

In [51]:
movies_data

movieId   imdbId    tmdbId                             title  \
1            2   113497    8844.0                           Jumanji   
18          19   112281    9273.0    Ace Ventura: When Nature Calls   
19          20   113845   11517.0                       Money Train   
31          32   114746      63.0                    Twelve Monkeys   
33          34   112431    9598.0                              Babe   
...        ...      ...       ...                               ...   
46894   176235   457437   14885.0  Pooh's Heffalump Halloween Movie   
46898   176249  5690142  404604.0                               Mom   
46905   176267   102797   30840.0                        Robin Hood   
46907   176271  2028550  111109.0               Century of Birthing   
46910   176279  6980792  461257.0                          Queerama   

      original_language                                           overview  \
1                    en  When siblings Judy and Peter discover an encha...   
18                   en  Summoned from an ashram in Tibet, Ace finds hi...   
19                   en  A vengeful New York transit cop decides to ste...   
31                   en  In the year 2035, convict James Cole reluctant...   
33                   en  Babe is a little pig who doesn't quite know hi...   
...                 ...                                                ...   
46894                en  It's Halloween in the 100 Acre Wood, and Roo's...   
46898                hi  The bliss of a biology teacher’s family life i...   
46905                en  Yet another version of the classic epic, with ...   
46907                tl  An artist struggles to finish his work while a...   
46910                en  50 years after decriminalisation of homosexual...   

       popularity  adult      budget  \
1       17.015539  False  65000000.0   
18       8.205448  False  30000000.0   
19       7.337906  False  60000000.0   
31      12.297305  False  29500000.0   
33      14.404764  False  30000000.0   
...           ...    ...         ...   
46894    2.568495  False         0.0   
46898    1.559596  False         0.0   
46905    5.683753  False         0.0   
46907    0.178241  False         0.0   
46910    0.163015  False         0.0   

                                                  genres  ... runtime  \
1      [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...  ...   104.0   
18     [{'id': 80, 'name': 'Crime'}, {'id': 35, 'name...  ...    90.0   
19     [{'id': 28, 'name': 'Action'}, {'id': 35, 'nam...  ...   103.0   
31     [{'id': 878, 'name': 'Science Fiction'}, {'id'...  ...   129.0   
33     [{'id': 14, 'name': 'Fantasy'}, {'id': 18, 'na...  ...    89.0   
...                                                  ...  ...     ...   
46894  [{'id': 16, 'name': 'Animation'}, {'id': 10751...  ...    67.0   
46898  [{'id': 80, 'name': 'Crime'}, {'id': 18, 'name...  ...   146.0   
46905  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...  ...   104.0   
46907                      [{'id': 18, 'name': 'Drama'}]  ...   360.0   
46910                                                 []  ...    75.0   

                                        spoken_languages    status  video  \
1      [{'iso_639_1': 'en', 'name': 'English'}, {'iso...  Released  False   
18              [{'iso_639_1': 'en', 'name': 'English'}]  Released  False   
19              [{'iso_639_1': 'en', 'name': 'English'}]  Released  False   
31     [{'iso_639_1': 'en', 'name': 'English'}, {'iso...  Released  False   
33              [{'iso_639_1': 'en', 'name': 'English'}]  Released  False   
...                                                  ...       ...    ...   
46894           [{'iso_639_1': 'en', 'name': 'English'}]  Released  False   
46898            [{'iso_639_1': 'hi', 'name': 'हिन्दी'}]  Released  False   
46905           [{'iso_639_1': 'en', 'name': 'English'}]  Released  False   
46907                  [{'iso_639_1': 'tl', 'name': ''}]  Released  False   
46910       

In [52]:
movies_data.shape

(6139, 24)

In [53]:
movies_data.isnull().sum()

movieId                 0
imdbId                  0
tmdbId                  0
title                   0
original_language       0
overview                0
popularity              0
adult                   0
budget                  0
genres                  0
production_companies    0
production_countries    0
release_date            0
revenue                 0
runtime                 0
spoken_languages        0
status                  0
video                   0
vote_average            0
vote_count              0
keywords                0
cast                    0
crew                    0
poster_working_url      0
dtype: int64

#### Final movies_data with no missing values shape(6139*26)

In [54]:
movies_data['popularity'].value_counts()

3.008299    8
0.000844    8
3.185256    8
0.000220    6
0.000308    6
           ..
0.619084    1
9.482806    1
0.661981    1
0.584232    1
0.163015    1
Name: popularity, Length: 5951, dtype: int64

In [55]:
movies_data['adult'].value_counts()

False    6135
True        4
Name: adult, dtype: int64

In [56]:
movies_data['budget'].value_counts()

0.0            4876
20000000.0       34
5000000.0        33
4000000.0        32
10000000.0       32
               ... 
2160000.0         1
5800000.0         1
60795000.0        1
258000000.0       1
3417000.0         1
Name: budget, Length: 370, dtype: int64

In [57]:
movies_data['genres'].value_counts()

[{'id': 18, 'name': 'Drama'}]                                                                                                 692
[{'id': 35, 'name': 'Comedy'}]                                                                                                463
[{'id': 99, 'name': 'Documentary'}]                                                                                           302
[]                                                                                                                            231
[{'id': 18, 'name': 'Drama'}, {'id': 10749, 'name': 'Romance'}]                                                               176
                                                                                                                             ... 
[{'id': 18, 'name': 'Drama'}, {'id': 10749, 'name': 'Romance'}, {'id': 36, 'name': 'History'}]                                  1
[{'id': 10769, 'name': 'Foreign'}, {'id': 35, 'name': 'Comedy'}]                          

In [58]:
movies_data['production_companies'].value_counts()

[]                                                                                                                                                                                                                                                                                  1372
[{'name': 'Metro-Goldwyn-Mayer (MGM)', 'id': 8411}]                                                                                                                                                                                                                                  120
[{'name': 'Twentieth Century Fox Film Corporation', 'id': 306}]                                                                                                                                                                                                                       78
[{'name': 'Warner Bros.', 'id': 6194}]                                                                                                                       

In [59]:
movies_data['production_countries'].value_counts()

[{'iso_3166_1': 'US', 'name': 'United States of America'}]                                                                                                                                     2349
[]                                                                                                                                                                                              633
[{'iso_3166_1': 'JP', 'name': 'Japan'}]                                                                                                                                                         277
[{'iso_3166_1': 'GB', 'name': 'United Kingdom'}]                                                                                                                                                276
[{'iso_3166_1': 'FR', 'name': 'France'}]                                                                                                                                                        251
                    

In [60]:
movies_data['release_date'].value_counts()

2005-09-03    17
2008-01-01    14
1981-01-01    13
2004-01-01    13
1989-01-01    13
              ..
2008-03-29     1
1960-12-01     1
1935-12-25     1
2003-04-10     1
2011-11-17     1
Name: release_date, Length: 4564, dtype: int64

In [61]:
movies_data['revenue'].value_counts()

0.0            5081
2500000.0         4
11000000.0        3
32000000.0        3
6000000.0         3
               ... 
940335536.0       1
51142659.0        1
259175788.0       1
6746000.0         1
184770205.0       1
Name: revenue, Length: 1021, dtype: int64

In [62]:
movies_data['runtime'].value_counts()

90.0     316
95.0     201
100.0    200
93.0     159
92.0     139
        ... 
252.0      1
925.0      1
237.0      1
207.0      1
34.0       1
Name: runtime, Length: 239, dtype: int64

In [63]:
movies_data['spoken_languages'].value_counts()

[{'iso_639_1': 'en', 'name': 'English'}]                                                                                                                                                                2902
[]                                                                                                                                                                                                       312
[{'iso_639_1': 'fr', 'name': 'Français'}]                                                                                                                                                                280
[{'iso_639_1': 'ja', 'name': '日本語'}]                                                                                                                                                                     258
[{'iso_639_1': 'ru', 'name': 'Pусский'}]                                                                                                                                            

In [64]:
movies_data['status'].value_counts()

Released           6064
Rumored              37
Post Production      29
In Production         5
Planned               4
Name: status, dtype: int64

In [65]:
movies_data['video'].value_counts()

False    6130
True        9
Name: video, dtype: int64

In [66]:
movies_data['vote_average'].value_counts()

6.0    318
7.0    272
0.0    258
6.5    249
5.0    230
      ... 
9.8      1
9.1      1
8.9      1
1.8      1
9.4      1
Name: vote_average, Length: 84, dtype: int64

In [67]:
movies_data['vote_count'].value_counts()

3.0       392
2.0       368
4.0       342
1.0       337
5.0       287
         ... 
8226.0      1
1946.0      1
1007.0      1
1410.0      1
324.0       1
Name: vote_count, Length: 740, dtype: int64

In [68]:
movies_data = movies_data.drop(columns = ['adult', 'video', 'status'] , axis = 1 )

In [69]:
movies_data

movieId   imdbId    tmdbId                             title  \
1            2   113497    8844.0                           Jumanji   
18          19   112281    9273.0    Ace Ventura: When Nature Calls   
19          20   113845   11517.0                       Money Train   
31          32   114746      63.0                    Twelve Monkeys   
33          34   112431    9598.0                              Babe   
...        ...      ...       ...                               ...   
46894   176235   457437   14885.0  Pooh's Heffalump Halloween Movie   
46898   176249  5690142  404604.0                               Mom   
46905   176267   102797   30840.0                        Robin Hood   
46907   176271  2028550  111109.0               Century of Birthing   
46910   176279  6980792  461257.0                          Queerama   

      original_language                                           overview  \
1                    en  When siblings Judy and Peter discover an encha...   
18                   en  Summoned from an ashram in Tibet, Ace finds hi...   
19                   en  A vengeful New York transit cop decides to ste...   
31                   en  In the year 2035, convict James Cole reluctant...   
33                   en  Babe is a little pig who doesn't quite know hi...   
...                 ...                                                ...   
46894                en  It's Halloween in the 100 Acre Wood, and Roo's...   
46898                hi  The bliss of a biology teacher’s family life i...   
46905                en  Yet another version of the classic epic, with ...   
46907                tl  An artist struggles to finish his work while a...   
46910                en  50 years after decriminalisation of homosexual...   

       popularity      budget  \
1       17.015539  65000000.0   
18       8.205448  30000000.0   
19       7.337906  60000000.0   
31      12.297305  29500000.0   
33      14.404764  30000000.0   
...           ...         ...   
46894    2.568495         0.0   
46898    1.559596         0.0   
46905    5.683753         0.0   
46907    0.178241         0.0   
46910    0.163015         0.0   

                                                  genres  \
1      [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
18     [{'id': 80, 'name': 'Crime'}, {'id': 35, 'name...   
19     [{'id': 28, 'name': 'Action'}, {'id': 35, 'nam...   
31     [{'id': 878, 'name': 'Science Fiction'}, {'id'...   
33     [{'id': 14, 'name': 'Fantasy'}, {'id': 18, 'na...   
...                                                  ...   
46894  [{'id': 16, 'name': 'Animation'}, {'id': 10751...   
46898  [{'id': 80, 'name': 'Crime'}, {'id': 18, 'name...   
46905  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...   
46907                      [{'id': 18, 'name': 'Drama'}]   
46910                                                 []   

                                    production_companies  ... release_date  \
1      [{'name': 'TriStar Pictures', 'id': 559}, {'na...  ...   1995-12-15   
18     [{'name': 'O Entertainment', 'id': 5682}, {'na...  ...   1995-11-10   
19              [{'name': 'Columbia Pictures', 'id': 5}]  ...   1995-11-21   
31     [{'name': 'Universal Pictures', 'id': 33}, {'n...  ...   1995-12-29   
33     [{'name': 'Universal Pictures', 'id': 33}, {'n...  ...   1995-07-18   
...                                                  ...  ...          ...   
46894        [{'name': 'Walt Disney Pictures', 'id': 2}]  ...   2005-09-13   
46898  [{'name': 'Mad Films', 'id': 82459}, {'name': ...  ...   2017-07-07   
46905  [{'name': 'Westdeutscher Rundfunk (WDR)', 'id'...  ...   1991-05-13   
46907             [{'name': 'Sine Olivia', 'id': 19653}]  ...   2011-11-17   
46910                                                 []  ...   2017-06-09   

           revenue  runtime  \
1      262797249.0    104.0   
18     212385533.0     90.0   
19      35431113.0    103.0   
31     168840000.0    129.0   
33     

In [70]:
movies_data.shape

(6139, 21)

In [71]:
movies_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6139 entries, 1 to 46910
Data columns (total 21 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   movieId               6139 non-null   int64  
 1   imdbId                6139 non-null   int64  
 2   tmdbId                6139 non-null   float64
 3   title                 6139 non-null   object 
 4   original_language     6139 non-null   object 
 5   overview              6139 non-null   object 
 6   popularity            6139 non-null   float64
 7   budget                6139 non-null   float64
 8   genres                6139 non-null   object 
 9   production_companies  6139 non-null   object 
 10  production_countries  6139 non-null   object 
 11  release_date          6139 non-null   object 
 12  revenue               6139 non-null   float64
 13  runtime               6139 non-null   float64
 14  spoken_languages      6139 non-null   object 
 15  vote_average        

In [72]:
movies_data.replace('[]', pd.NA, inplace=True)

# Drop rows with NaN values in any column
movies_data.dropna(inplace=True)

In [73]:
movies_data

movieId   imdbId    tmdbId                             title  \
1            2   113497    8844.0                           Jumanji   
18          19   112281    9273.0    Ace Ventura: When Nature Calls   
19          20   113845   11517.0                       Money Train   
31          32   114746      63.0                    Twelve Monkeys   
33          34   112431    9598.0                              Babe   
...        ...      ...       ...                               ...   
46878   176201   232750    5589.0           Šíleně smutná princezna   
46884   176215    36975   84710.0                      Jungle Woman   
46894   176235   457437   14885.0  Pooh's Heffalump Halloween Movie   
46898   176249  5690142  404604.0                               Mom   
46907   176271  2028550  111109.0               Century of Birthing   

      original_language                                           overview  \
1                    en  When siblings Judy and Peter discover an encha...   
18                   en  Summoned from an ashram in Tibet, Ace finds hi...   
19                   en  A vengeful New York transit cop decides to ste...   
31                   en  In the year 2035, convict James Cole reluctant...   
33                   en  Babe is a little pig who doesn't quite know hi...   
...                 ...                                                ...   
46878                cs                                 No overview found.   
46884                en  Paula the ape woman (Acquanetta) is alive and ...   
46894                en  It's Halloween in the 100 Acre Wood, and Roo's...   
46898                hi  The bliss of a biology teacher’s family life i...   
46907                tl  An artist struggles to finish his work while a...   

       popularity      budget  \
1       17.015539  65000000.0   
18       8.205448  30000000.0   
19       7.337906  60000000.0   
31      12.297305  29500000.0   
33      14.404764  30000000.0   
...           ...         ...   
46878    0.375001         0.0   
46884    0.143223         0.0   
46894    2.568495         0.0   
46898    1.559596         0.0   
46907    0.178241         0.0   

                                                  genres  \
1      [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
18     [{'id': 80, 'name': 'Crime'}, {'id': 35, 'name...   
19     [{'id': 28, 'name': 'Action'}, {'id': 35, 'nam...   
31     [{'id': 878, 'name': 'Science Fiction'}, {'id'...   
33     [{'id': 14, 'name': 'Fantasy'}, {'id': 18, 'na...   
...                                                  ...   
46878  [{'id': 10402, 'name': 'Music'}, {'id': 35, 'n...   
46884  [{'id': 18, 'name': 'Drama'}, {'id': 14, 'name...   
46894  [{'id': 16, 'name': 'Animation'}, {'id': 10751...   
46898  [{'id': 80, 'name': 'Crime'}, {'id': 18, 'name...   
46907                      [{'id': 18, 'name': 'Drama'}]   

                                    production_companies  ... release_date  \
1      [{'name': 'TriStar Pictures', 'id': 559}, {'na...  ...   1995-12-15   
18     [{'name': 'O Entertainment', 'id': 5682}, {'na...  ...   1995-11-10   
19              [{'name': 'Columbia Pictures', 'id': 5}]  ...   1995-11-21   
31     [{'name': 'Universal Pictures', 'id': 33}, {'n...  ...   1995-12-29   
33     [{'name': 'Universal Pictures', 'id': 33}, {'n...  ...   1995-07-18   
...                                                  ...  ...          ...   
46878  [{'name': 'Filmové Studio Barrandov', 'id': 19...  ...   1968-06-07   
46884         [{'name': 'Universal Pictures', 'id': 33}]  ...   1944-06-01   
46894        [{'name': 'Walt Disney Pictures', 'id': 2}]  ...   2005-09-13   
46898  [{'name': 'Mad Films', 'id': 82459}, {'name': ...  ...   2017-07-07   
46907             [{'name': 'Sine Olivia', 'id': 19653}]  ...   2011-11-17   

           revenue  runtime  \
1      262797249.0    104.0   
18     212385533.0     90.0   
19      35431113.0    103.0   
31     168840000.0    129.0   
33     

**Final movies data with 3660 rows and 21 columns**

1. Removed columns - ['adult', 'video', 'status'] in Movies_data(3660) and ['belongs_to_collection', 'homepage', 'tagline', 'imdb_id', 'poster_path', 'original_title'] in Movies_data(6139). 

2. In Movies_data(6139), all missing values are removed except the empty lists present for 'spoken_languages', 'production_countries', 'production_companies', 'genres', 'cast', 'crew' and 'keywords'.

3. In Movies_data(3660) all mising values are removed including empty lists.

4. Dim(Movies_data(3660)) = 3660x21

5. Dim(Movies_data(6139)) = 6139x24

In [74]:
movies_data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3660 entries, 1 to 46907
Data columns (total 21 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   movieId               3660 non-null   int64  
 1   imdbId                3660 non-null   int64  
 2   tmdbId                3660 non-null   float64
 3   title                 3660 non-null   object 
 4   original_language     3660 non-null   object 
 5   overview              3660 non-null   object 
 6   popularity            3660 non-null   float64
 7   budget                3660 non-null   float64
 8   genres                3660 non-null   object 
 9   production_companies  3660 non-null   object 
 10  production_countries  3660 non-null   object 
 11  release_date          3660 non-null   object 
 12  revenue               3660 non-null   float64
 13  runtime               3660 non-null   float64
 14  spoken_languages      3660 non-null   object 
 15  vote_average        